test/EndToEndTests/Tests/Client/Microsoft.OData.Client.E2E.Tests/EdmDateAndTimeOfDayTests/Tests/EdmDateAndTimeOfDayTests.cs

```diff
@@ -0,0 +1,501 @@
+            var responseMessage = await requestMessage.GetResponseAsync();
+
+            Assert.Equal(200, responseMessage.StatusCode);
+
+            using var messageReader = new ODataMessageReader(responseMessage, readerSettings, _model);
+            var True = (await messageReader.ReadPropertyAsync()).Value;
+
+            Assert.Equal(false, True);
+        }
+
+        [Theory]
+        [MemberData(nameof(MimeTypesData))]
+        public async Task QueryingByDateKey_WorksCorrectly(string mimeType)
+        {
+            ODataMessageReaderSettings readerSettings = new() { BaseUri = _baseUri };
+
+            var requestUrl = new Uri(_baseUri.AbsoluteUri + "Calendars(2015-11-11)", UriKind.Absolute);
+
+            var requestMessage = new TestCommon.Common.HttpClientRequestMessage(requestUrl, Client);
+
+            requestMessage.SetHeader("Accept", mimeType);
+            var responseMessage = await requestMessage.GetResponseAsync();
+
+            Assert.Equal(200, responseMessage.StatusCode);
+
+            if (!mimeType.Contains(MimeTypes.ODataParameterNoMetadata))
+            {
+                using var messageReader = new ODataMessageReader(responseMessage, readerSettings, _model);
+                var reader = await messageReader.CreateODataResourceReaderAsync();
+
+                while (await reader.ReadAsync())
+                {
+                    if (reader.State == ODataReaderState.ResourceEnd)
+                    {
+                        ODataResource entry = reader.Item as ODataResource;
+                        // Verify Date Property
+                        Assert.Equal(new Date(2015, 11, 11), entry.Properties.OfType<ODataProperty>().Single(p => p.Name == "Day").Value);
+                    }
+                }
+
+                Assert.Equal(ODataReaderState.Completed, reader.State);
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(MimeTypesData))]
+        public async Task InvokingAnActionWithDateAndTimeParameters_WorksCorrectly(string mimeType)
+        {
+            var writerSettings = new ODataMessageWriterSettings
+            {
+                BaseUri = _baseUri,
+                EnableMessageStreamDisposal = false
+            };
+
+            var readerSettings = new ODataMessageReaderSettings
+            {
+                BaseUri = _baseUri
+            };
+
+            var requestUrl = new Uri(_baseUri + "Orders(7)/Default.ChangeShipTimeAndDate");
+
+            var requestMessage = new TestCommon.Common.HttpClientRequestMessage(requestUrl, Client)
+            {
+                Method = "POST"
+            };
+
+            requestMessage.SetHeader("Content-Type", mimeType);
+            requestMessage.SetHeader("Accept", mimeType);
+
+            Date newDate = Date.MinValue;
+            TimeOfDay newTime = TimeOfDay.MinValue;
+
+            using (var messageWriter = new ODataMessageWriter(requestMessage, writerSettings, _model))

```

Comment:
```suggestion
            await using (var messageWriter = new ODataMessageWriter(requestMessage, writerSettings, _model))
```
We should make it a practice to `await using` in async tests where the type implements `IAsyncDisposable`. That way the async dispoable logic get good test coverage too